frontend/projects/upgrade/src/app/features/dashboard/segments/components/segments-list/segments-list.component.ts

```diff
@@ -69,13 +79,45 @@ export class SegmentsListComponent implements OnInit, OnDestroy, AfterViewInit {
     this.permissions$ = this.authService.userPermissions$;
     this.allSegmentsSub = this.segmentsService.allSegments$.subscribe((segments) => {
       segments = segments.map((segment) => ({ ...segment, status: segment.status || SEGMENT_STATUS.UNUSED }));
-      this.allSegments = new CustomMatTableSource();
+      this.allSegments = new MatTableDataSource();
       this.allSegments.data = [...segments];
       this.allSegments.sort = this.sort;
       this.cdr.detectChanges();
+      this.applyFilter(this.searchValue);
     });
   }
 
+  filterExperimentPredicate(type: EXPERIMENT_SEARCH_KEY) {
+    this.allSegments.filterPredicate = (data, filter: string): boolean => {
+      switch (type) {
+        case EXPERIMENT_SEARCH_KEY.ALL:
+          return (
+            data.name.toLocaleLowerCase().includes(filter) ||
+            data.status.toLocaleLowerCase().includes(filter) ||
+            data.context.toLocaleLowerCase().includes(filter)
+          );
+        case EXPERIMENT_SEARCH_KEY.NAME:
+          return data.name.toLocaleLowerCase().includes(filter);
+        case EXPERIMENT_SEARCH_KEY.STATUS:
+          return data.status.toLocaleLowerCase().includes(filter);
+        case EXPERIMENT_SEARCH_KEY.CONTEXT:
+          return data.context.toLocaleLowerCase().includes(filter);
+      }
+    };
+  }
+
+  applyFilter(filterValue: string) {
+    this.filterExperimentPredicate(this.selectedSegmentFilterOption);
+    if (typeof filterValue === 'string') {
+      this.allSegments.filter = filterValue.trim().toLowerCase();
+    }
+  }
+
+  filterExperimentByChips(tagValue: string, type: EXPERIMENT_SEARCH_KEY) {
```

Comment:
This method is not used. If we want to use it, we should add the actions to the template for 'context' and 'tags', the same as in the experiment template, and also change its name to `filterSegmentByChips()`

Reply:
removed that func for now, done

Reply:
I want to check with product to make sure we don't want to have the filter by chaps functionality. I suspect we probably do.

Reply:
I am not sure, if we do and if you think we won't have to redo same in new design can you create separate ticket as this PR target issue of search functionality not working. We had discussed same in our meeting, and there were few other functionality I can add but since we are about start for new design and issue was about search functionality not working so address that only in this PR.